CAPI/cpp/API/include/API.h

```diff
@@ -41,6 +41,13 @@ class ILogic
     virtual std::vector<std::vector<THUAI6::PlaceType>> GetFullMap() const = 0;
     virtual THUAI6::PlaceType GetPlaceType(int32_t cellX, int32_t cellY) const = 0;
 
+    virtual int32_t GetClassroomProgress(int32_t cellX, int32_t cellY) const = 0;
+    virtual int32_t GetChestProgress(int32_t cellX, int32_t cellY) const = 0;
+    virtual int32_t GetGateProgress(int32_t cellX, int32_t cellY) const = 0;
+    virtual bool IsDoorOpen(int32_t cellX, int32_t cellY) const = 0;
+
+    virtual std::shared_ptr<const THUAI6::GameInfo> GetGameInfo() const = 0;
```

Comment:
```suggestion
    [[nodiscard]] virtual std::shared_ptr<const THUAI6::GameInfo> GetGameInfo() const = 0;
```

此处以上所有 `Get` 函数都加上 `[[nodiscard]]` 吧